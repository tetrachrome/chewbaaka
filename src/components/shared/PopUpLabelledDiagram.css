/**
 * PopUpLabelledDiagram.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 16, 2020
 * Updated  : Jul 16, 2020
 */

div.PopUpLabelledDiagramOuterComponent {
  background-size: cover;
}

div.PopUpLabelledDiagramOuterComponent_DebugColor {
  background-color: slateblue;
}

div.PopUpLabelledDiagramInnerComponent {
  position: absolute;
}

div.PopUpLabelledDiagramInnerComponent_DebugColor {
  background-color: rosybrown;
}

div.PopUpLabelledDiagramPopUpItemComponent {
  position: absolute;
  display: inline-block;
}

div.PopUpLabelledDiagramPopUpItemComponent_DebugColor {
  background-color: palegoldenrod;
}

h4.PopUpLabelledDiagramPopUpItemTitle {
  font-size: 20px;
  padding-bottom: 10px;
  border-bottom: 2px solid teal;
}
