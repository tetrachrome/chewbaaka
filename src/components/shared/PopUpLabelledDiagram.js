/**
 * PopUpLabelledDiagram.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 16, 2020
 * Updated  : Jul 16, 2020
 */

import React from 'react'

import { Button, Popup } from 'semantic-ui-react'

import { getElementStyleClassName } from '../../styling/styling'

import './PopUpLabelledDiagram.css'

export default class PopUpLabelledDiagram extends React.Component {

  render() {
    const componentDimensionStyle = {
      width: this.props.width,
      height: this.props.height,
      backgroundImage: `url(${this.props.image})`,
    };

    const popups = this.props.items.map(this.renderPopUp);

    return (
      <div
        style={componentDimensionStyle}
        className={getElementStyleClassName("PopUpLabelledDiagramOuterComponent")}
      >
        <div className={getElementStyleClassName("PopUpLabelledDiagramInnerComponent")}>
          {popups}
        </div>
      </div>
    );
  }

  renderPopUp(item) {
    const popUpContainerPositionStyle = item.position;

    return (
      <div style={popUpContainerPositionStyle} className={getElementStyleClassName("PopUpLabelledDiagramPopUpItemComponent")}>
        <Popup
          size="large"
          trigger={<Button icon="plus" color="blue" circular size="mini" />}
          content={item.content}
        />
      </div>
    );
  }
}
