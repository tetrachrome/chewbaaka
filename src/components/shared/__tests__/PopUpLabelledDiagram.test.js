/**
 * PopUpLabelledDiagram.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 16, 2020
 * Updated  : Aug 05, 2020
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import PopUpLabelledDiagram from '../PopUpLabelledDiagram'

import image from '../assets/cheetah-conservation-fund-logo-mini-min.jpg'

const __ITEMS__ = [
  {
    title: "Title 1",
    content: "Hello. This is a regular pop-up which does not allow for lots of content. You cannot fit a lot of words here as the paragraphs will be pretty narrow.",
    position: {
      top: 100,
      left: 30
    }
  },
  {
    title: "Title 2",
    content: "This is another item"
  }
];

test('renders PopUpLabelledDiagram component', () => {
  render(
    <PopUpLabelledDiagram
      image={image}
      width={1814}
      height={1043}
      items={__ITEMS__}
    />
  );
});

test('PopUpLabelledDiagram component snapshot', () => {
  const tree = renderer
    .create(
      <PopUpLabelledDiagram
        image={image}
        width={1814}
        height={1043}
        items={__ITEMS__}
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
