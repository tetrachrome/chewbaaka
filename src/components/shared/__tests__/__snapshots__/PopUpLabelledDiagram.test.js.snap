// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUpLabelledDiagram component snapshot 1`] = `
<div
  className="PopUpLabelledDiagramOuterComponent"
  style={
    Object {
      "backgroundImage": "url(cheetah-conservation-fund-logo.jpg)",
      "height": 1043,
      "width": 1814,
    }
  }
>
  <div
    className="PopUpLabelledDiagramInnerComponent"
  >
    <div
      className="PopUpLabelledDiagramPopUpItemComponent"
      style={
        Object {
          "left": 30,
          "top": 100,
        }
      }
    >
      <button
        className="ui blue mini circular icon button"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="plus icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="PopUpLabelledDiagramPopUpItemComponent"
    >
      <button
        className="ui blue mini circular icon button"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <i
          aria-hidden="true"
          className="plus icon"
          onClick={[Function]}
        />
      </button>
    </div>
  </div>
</div>
`;
